runtime/hydradx/src/xcm.rs

```diff
@@ -0,0 +1,279 @@
+			)),
+			_ => {
+				if let Some(loc) = AssetRegistry::asset_to_location(id) {
+					Some(loc.0)
+				} else {
+					None
+				}
+			}
+		}
+	}
+}
+
+impl Convert<MultiLocation, Option<AssetId>> for CurrencyIdConvert {
+	fn convert(location: MultiLocation) -> Option<AssetId> {
+		match location {
+			MultiLocation {
+				parents,
+				interior: X2(Parachain(id), GeneralKey(key)),
+			} if parents == 1 && ParaId::from(id) == ParachainInfo::get() => {
+				// Handling native asset for this parachain
+				if let Ok(currency_id) = AssetId::decode(&mut &key[..]) {
+					// we currently have only one native asset
+					match currency_id {
+						CORE_ASSET_ID => Some(currency_id),
+						_ => None,
+					}
+				} else {
+					None
+				}
+			}
+			// handle reanchor canonical location: https://github.com/paritytech/polkadot/pull/4470
+			MultiLocation {
+				parents: 0,
+				interior: X1(GeneralKey(key)),
+			} => {
+				if let Ok(currency_id) = AssetId::decode(&mut &key[..]) {
+					// we currently have only one native asset
+					match currency_id {
+						CORE_ASSET_ID => Some(currency_id),
+						_ => None,
+					}
+				} else {
+					None
+				}
+			}
+			// delegate to asset-registry
+			_ => AssetRegistry::location_to_asset(AssetLocation(location)),
+		}
+	}
+}
+
+impl Convert<MultiAsset, Option<AssetId>> for CurrencyIdConvert {
+	fn convert(asset: MultiAsset) -> Option<AssetId> {
+		if let MultiAsset {
+			id: Concrete(location), ..
+		} = asset
+		{
+			Self::convert(location)
+		} else {
+			None
+		}
+	}
+}
+
+pub struct AccountIdToMultiLocation;
+impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
+	fn convert(account: AccountId) -> MultiLocation {
+		X1(AccountId32 {
+			network: NetworkId::Any,
+			id: account.into(),
+		})
+		.into()
+	}
+}
+
+/// The means for routing XCM messages which are not for local execution into the right message
+/// queues.
+pub type XcmRouter = (
+	// Two routers - use UMP to communicate with the relay chain:
+	cumulus_primitives_utility::ParentAsUmp<ParachainSystem, ()>,
+	// ..and XCMP to communicate with the sibling chains.
+	XcmpQueue,
+);
+
+/// Type for specifying how a `MultiLocation` can be converted into an `AccountId`. This is used
+/// when determining ownership of accounts for asset transacting and when attempting to use XCM
+/// `Transact` in order to determine the dispatch Origin.
+pub type LocationToAccountId = (
+	// The parent (Relay-chain) origin converts to the default `AccountId`.
+	ParentIsDefault<AccountId>,
+	// Sibling parachain origins convert to AccountId via the `ParaId::into`.
+	SiblingParachainConvertsVia<Sibling, AccountId>,
+	// Straight up local `AccountId32` origins just alias directly to `AccountId`.
+	AccountId32Aliases<RelayNetwork, AccountId>,
+);
+
+pub type LocalAssetTransactor = MultiCurrencyAdapter<
+	Currencies,
+	UnknownTokens,
+	IsNativeConcrete<AssetId, CurrencyIdConvert>,
+	AccountId,
+	LocationToAccountId,
+	AssetId,
+	CurrencyIdConvert,
+	(),
```

Comment:
Add the same error deposit handler as done in basilisk.

Reply:
Ok. I will update it once the PR with deposit handler is closed.